tests/unit/lms/services/h_api_test.py

```diff
@@ -83,83 +77,64 @@ def test_upsert_calls_calls_create_if_update_fails(
         h_api.update_group(sentinel.group_id, sentinel.group_name)
 
-        h_api._api_request.assert_called_once_with(
+        _api_request.assert_called_once_with(
             "PATCH", "groups/sentinel.group_id", data={"name": sentinel.group_name}
         )
 
-    def test_add_user_to_group(self, h_api, h_user):
+    def test_add_user_to_group(self, h_api, _api_request, h_user):
         h_api.add_user_to_group(h_user, sentinel.group_id)
 
-        h_api._api_request.assert_called_once_with(
+        _api_request.assert_called_once_with(
             "POST",
-            "groups/sentinel.group_id/members/acct:sentinel.username@test-authority",
-        )
-
-    @pytest.fixture
-    def h_user(self):
-        return HUser(
-            username=sentinel.username,
-            display_name=sentinel.display_name,
-            authority=AUTHORITY,
+            "groups/sentinel.group_id/members/acct:sentinel.username@TEST_AUTHORITY",
         )
 
-    @pytest.yield_fixture
-    def h_api(self, p_request):
-        h_api = HAPI(sentinel.context, p_request)
-
-        with patch.object(HAPI, "_api_request"):
-            yield h_api
-
-
-class TestHAPIRequest:
-
-    # We're accessing h_api._api_request a lot in this test class, so disable
-    # protected-access messages.
-    # pylint: disable=protected-access
-
-    def test_it_passes_expected_defaults(self, h_api, requests):
+    def test__api_request(self, h_api, requests):
         h_api._api_request(sentinel.method, "dummy-path")
 
-        self._assert_called_requests_with(
-            requests, auth=(sentinel.client_id, sentinel.client_secret), timeout=10
-        )
-
-    def test_it_passes_method_and_path(self, h_api, requests):
-        h_api._api_request(sentinel.method, "/path")
-
-        self._assert_called_requests_with(
-            requests, method=sentinel.method, url=BASE_URL + "path",
-        )
-
-    def test_id_dumps_json_body(self, h_api, requests):
+        assert requests.request.call_args_list == [
+            call(
+                # It uses the given HTTP request method.
+                method=sentinel.method,
+                # It requests the given URL.
+                url="https://example.com/private/api/dummy-path",
+                # It adds the authentication headers.
+                auth=("TEST_CLIENT_ID", "TEST_CLIENT_SECRET"),
+                # Requests time out after 10 seconds.
+                timeout=10,
+                # It adds the Hypothesis-Application header.
+                headers={"Hypothesis-Application": "lms"},
+            )
+        ]
+
+    def test_if_given_a_data_dict__api_request_dumps_it_to_a_json_string(
+        self, h_api, requests
+    ):
         h_api._api_request(sentinel.method, "dummy-path", {"a": 1, "b": [2]})
 
-        self._assert_called_requests_with(requests, data='{"a":1,"b":[2]}')
+        assert requests.request.call_args[1]["data"] == '{"a":1,"b":[2]}'
```

Comment:
Replacing the `self._assert_called_requests_with()` helper with normal mock library usage